src/cmap/auth/mongo_credentials.ts

```diff
@@ -30,24 +30,16 @@ function getDefaultAuthMechanism(hello?: Document): AuthMechanism {
   return AuthMechanism.MONGODB_CR;
 }
 
-/**
- * TODO: NODE-5035: Make OIDC properties public.
- *
- * @public
- * */
+/** @public */
 export interface AuthMechanismProperties extends Document {
   SERVICE_HOST?: string;
   SERVICE_NAME?: string;
   SERVICE_REALM?: string;
   CANONICALIZE_HOST_NAME?: GSSAPICanonicalizationValue;
   AWS_SESSION_TOKEN?: string;
-  /** @internal Name for the OIDC device workflow */
   DEVICE_NAME?: 'aws' | 'azure' | 'gcp';
-  /** @internal Similar to a username, is require by OIDC when more than one IDP is configured. */
   PRINCIPAL_NAME?: string;
-  /** @internal User provided callback to get OIDC auth credentials */
   REQUEST_TOKEN_CALLBACK?: OIDCRequestFunction;
-  /** @internal User provided callback to refresh OIDC auth credentials */
   REFRESH_TOKEN_CALLBACK?: OIDCRefreshFunction;
```

Comment:
Also fyi, these functions now also take the principal name (i.e. username passed to the initial MongoClient) as an argument, as well as a timeout mechanism. In the spec, the example signature uses an integer, but referring to the spec text

> This can be as a timeout argument to the callback, a cancellation context passed to the callback, or some other language-appropriate mechanism. 

I would say that the “language-appropriate mechanism” for this is an `AbortSignal`.

Reply:
Yeah I just saw the spec was updated there. Thanks for the recommendation!

Reply:
Updated.